src/main/index.js

```diff
@@ -1550,6 +1559,15 @@ function runApp() {
     }
   }
 
+  function macOSnewURLWindow(url) {
+    app.once('browser-window-created', (_, mainWindow) => {
+      mainWindow.webContents.once('did-finish-load', () => {
+        // A timeout here is necessary or the new window won't receive the URL
+        setTimeout(function() { mainWindow.webContents.send(IpcChannels.OPEN_URL, url, { isLaunchLink: true }) }, 1000)
+      })
+    })
```

Comment:
If you reuse the same code that is used at app-startup you avoid having to hardcode a timeout (generally a good idea to check the existing code first before re-inventing the wheel).

https://github.com/FreeTubeApp/FreeTube/blob/9e09ec5dc6d2dacbc1cc8a9143e0ccce71e1a750/src/main/index.js#L832-L836

```suggestion
    ipcMain.once(IpcChannels.APP_READY, () => {
      mainWindow.webContents.send(IpcChannels.OPEN_URL, URL, { isLaunchLink: true })
    })
```

Reply:
My bad, I've committed changes so that this callback will be used instead.